src/shared/wizards/wizardForm.ts

```diff
@@ -3,87 +3,159 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Prompter } from '../ui/prompter'
 import * as _ from 'lodash'
-import { StateWithCache, WizardState } from './wizard'
-import { ExpandWithObject } from '../utilities/tsUtils'
+import * as toposort from 'toposort'
+import { Prompter } from '../ui/prompter'
+import { StateWithCache } from './wizard'
+import { ExpandWithObject, ReduceTuple } from '../utilities/tsUtils'
+
+export type PrompterProvider<TState, TProp, TDep> = (
+    state: StateWithCache<BoundState<TState, TDep>, TProp>
+) => Prompter<TProp>
+export type BoundState<TState, Bindings> = FindPartialTuple<TState, ExtractPaths<Bindings>>
```

Comment:
Lots of explanation about the `FindPartialTuple` but there's no actual description for what this does with that information and how to use this type, despite the fact that this is the only part that future users will see.